File: easy/1957_delete_characters_to_make_fancy_string/solution.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        fancy_s = []

        prev = None
        prev_count = 0

        for c in s:
            if c == prev:
                prev_count += 1

                if prev_count >= 3:
                    continue
            else:
                prev = c
                prev_count = 1

            fancy_s.append(c)

        return "".join(fancy_s)

File: easy/1957_delete_characters_to_make_fancy_string/solution.cpp
using namespace std;

#include <string>

class Solution {
public:
  string makeFancyString(string s) {
    string fancyS = "";
    fancyS.reserve(s.size());

    char prev;
    int prevCount = 0;

    for (const char &c : s) {
      if (c == prev) {
        if (++prevCount > 2)
          continue;
      } else {
        prev = c;
        prevCount = 1;
      }

      fancyS.push_back(c);
    }

    return fancyS;
  }
};
